orca-webhook/src/main/java/com/netflix/spinnaker/orca/webhook/config/WebhookConfiguration.java

```diff
@@ -116,58 +125,123 @@ public ClientHttpRequestFactory webhookRequestFactory(
+  private SSLSocketFactory getSSLSocketFactory(X509TrustManager trustManager) throws IOException {
     try {
-      SSLContext sslContext = SSLContext.getInstance("TLS");
-      sslContext.init(null, new X509TrustManager[] {trustManager}, null);
-      return sslContext.getSocketFactory();
+      var identityOpt = getCustomIdentity();
+      if (identityOpt.isPresent()) {
+        var identity = identityOpt.get();
+        return identity.createSSLContext(trustManager).getSocketFactory();
+      } else {
+        var sslContext = SSLContext.getInstance("TLS");
+        sslContext.init(null, new X509TrustManager[] {trustManager}, null);
+        return sslContext.getSocketFactory();
+      }
     } catch (KeyManagementException | NoSuchAlgorithmException e) {
       throw new RuntimeException(e);
     }
   }
 
   private X509TrustManager getTrustManager(KeyStore keyStore) {
     try {
-      TrustManagerFactory trustManagerFactory =
+      var trustManagerFactory =
           TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
       trustManagerFactory.init(keyStore);
-      TrustManager[] trustManagers = trustManagerFactory.getTrustManagers();
+      var trustManagers = trustManagerFactory.getTrustManagers();
       return (X509TrustManager) trustManagers[0];
     } catch (KeyStoreException | NoSuchAlgorithmException e) {
       throw new RuntimeException(e);
     }
   }
 
-  private Optional<KeyStore> getCustomKeyStore() {
-    WebhookProperties.TrustSettings trustSettings = webhookProperties.getTrust();
-    if (trustSettings == null
-        || !trustSettings.isEnabled()
-        || StringUtils.isEmpty(trustSettings.getTrustStore())) {
+  private Optional<X509Identity> getCustomIdentity() throws IOException {
+    var identitySettings = webhookProperties.getIdentity();
+    if (identitySettings == null || !identitySettings.isEnabled()) {
       return Optional.empty();
     }
 
-    KeyStore keyStore;
-    try {
-      keyStore = KeyStore.getInstance(KeyStore.getDefaultType());
-    } catch (KeyStoreException e) {
-      throw new RuntimeException(e);
+    if (StringUtils.isNotEmpty(identitySettings.getIdentityStore())) {
+      var identity =
+          X509IdentitySource.fromKeyStore(
+              Path.of(identitySettings.getIdentityStore()),
+              identitySettings.getIdentityStoreType(),
+              () -> {
+                var password = identitySettings.getIdentityStorePassword();
+                return password == null ? new char[0] : password.toCharArray();
+              },
+              () -> {
+                var password = identitySettings.getIdentityStoreKeyPassword();
+                return password == null ? new char[0] : password.toCharArray();
+              });
+      return Optional.of(identity.load());
+    } else if (StringUtils.isNotEmpty(identitySettings.getIdentityKeyPem())) {
+      return Optional.of(
+          X509IdentitySource.fromPEM(
```

Comment:
Constructs a client identity from PEM key and PEM cert.  No encryption or password-protection supported.